daml/Common.daml

```diff
@@ -128,12 +121,35 @@ onboardCustomer CommonContracts{..} party asset accountNominee = do
   identityVerificationRequestCid <- submit customer do exerciseCmd regulatorServiceCid Regulator.RequestIdentityVerification with observers = [public]; legalName = party <> " Legal Name"; location = party <> " Location"
   verifiedIdentity <- submit operator do exerciseCmd regulatorServiceCid Regulator.VerifyIdentity with ..
 
-  -- Assets
-  creditAccountRequestCid <- submit customer do exerciseCmd custodyServiceCid Custody.RequestCreditAccount with accountId = tradingAccountId; ..
-  assetDepositCid <- submit custodian do exerciseCmd custodyServiceCid Custody.CreditAccount with ..
-
   -- Bidding Service
-  biddingServiceOfferCid <- submit agent do exerciseCmd distributorRoleCid Distributor.OfferBiddingService with ..
-  biddingServiceCid <- submit customer do exerciseCmd biddingServiceOfferCid Bidding.Accept with ..
+  biddingServiceOfferCid <- submit bank do exerciseCmd distributorRoleCid Distributor.OfferBiddingService with ..
+  biddingServiceCid <- submit customer do exerciseCmd biddingServiceOfferCid Bidding.Accept with tradingAccount = mainAccount; allocationAccount = auctionLockedAccount
+
+  pure $ Customer with ..
 
-  pure $ CustomerContracts with ..
+onboardAssets : Providers -> Script Assets
+onboardAssets Providers{..} = do
+  let
+    safekeepingAccountId = Id with signatories = fromList [bank]; label = "BankSafekeepingAccount"; version = 0
+    observers = singleton public
+    usdId = Id with signatories = fromList [ bank ]; label = "USD"; version = 0
+    usd = Asset with id = usdId; quantity = 0.0
+    tslaId = Id with signatories = singleton bank; label = "TSLA"; version = 0
+    tsla = Asset with id = tslaId; quantity = 0.0
+    nflxId = Id with signatories = singleton bank; label = "NFLX"; version = 0
+    nflx = Asset with id = nflxId; quantity = 0.0
+    dUnitId = Id with signatories = singleton bank; label = "DAH-D"; version = 0
+    dUnit = Asset with id = nflxId; quantity = 0.0
+
+  submit bank do createCmd AssetDescription with assetId = usdId; description = "United States Dollar"; issuer = bank; claims = Zero; ..
+  submit bank do createCmd AssetDescription with assetId = nflxId; description = "Netflix, Inc. Common Stock"; issuer = bank; claims = Zero; ..
+  submit bank do createCmd AssetDescription with assetId = tslaId; description = "Tesla, Inc. Common Stock"; issuer = bank; claims = Zero; ..
+  submit bank do createCmd AssetDescription with assetId = dUnitId; description = "Digital Asset Holdings D-Unit"; issuer = bank; claims = Zero; ..
```

Comment:
Don't you have problems here because the custodian isn't a signatory? Why not use `RequestOrigination` with `claims = Zero` instead?